### PR TITLE
IDEA-277826 Do not show 'Run Tests' option when no tests available to run

### PR DESCRIPTION
**Pull request by karimai-gg**
when creating configurations for AllInDirectoryGradleConfigurationProducer, take into consideration the fact of actual test sources being available within the given context location.

### CODE REVIEW

File: plugins/gradle/java/src/execution/test/runner/AllInPackageGradleConfigurationProducer.java [hunk 1]
```diff
@@ -73,7 +84,12 @@ protected boolean doIsConfigurationFromContext(ExternalSystemRunConfiguration co
 
     final String scriptParameters = configuration.getSettings().getScriptParameters() + ' ';
     final String filter = createTestFilterFrom(configurationData.psiPackage, /*hasSuffix=*/true);
-    return scriptParameters.contains(filter);
+    if (!scriptParameters.contains(filter)) return false;
+    VirtualFile locationFile = (VirtualFile)configuration.getUserData(LOCATION_NAME);
+    // If locationFile is null, then return true. This is for all the existing configurations created without location information
+    // from previous versions, as they would have already passed all the previous comparisons.
+    if (locationFile == null) return true;
```
Comment (nskvortsov) [inline plugins/gradle/java/src/execution/test/runner/AllInPackageGradleConfigurationProducer.java:91]: I am not sure why you would like to match run configurations based on `locationFile`, but please be aware that `get/setUserData`  is not persistent and may have desired effect after IDEA restart.

**Review by nskvortsov — CHANGES_REQUESTED** [review]
Please add a test, displaying how this feature should work. You can take a look at `org.jetbrains.plugins.gradle.execution.test.runner.GradleTestRunConfigurationProducerTest` as an example.

**Review by DeveloperHacker — CHANGES_REQUESTED** [review]

File: plugins/gradle/java/src/execution/test/runner/AllInPackageGradleConfigurationProducer.java (hunk 1 above)
Comment (karimai-gg) [reply plugins/gradle/java/src/execution/test/runner/AllInPackageGradleConfigurationProducer.java]: Please ignore that. I forgot to clean a very old commit. Th PR is just about a small change now. PTAL and let me know if looks good !

[the remark this answers is not in the document]
Comment (karimai-gg) [reply plugins/gradle/java/src/execution/test/runner/AllInPackageGradleConfigurationProducer.java]: This was not supposed to be here (it's a very old commit that was kept)

[the remark this answers is not in the document]
Comment (DeveloperHacker) [reply plugins/gradle/java/src/execution/test/runner/AllInPackageGradleConfigurationProducer.java]: Could you create new pull request with one commit?

[the remark this answers is not in the document]
Comment (karimai-gg) [reply plugins/gradle/java/src/execution/test/runner/AllInPackageGradleConfigurationProducer.java]: is it ok to abandon this PR and add same reviewers in the new one ?

[the remark this answers is not in the document]
Comment (DeveloperHacker) [reply plugins/gradle/java/src/execution/test/runner/AllInPackageGradleConfigurationProducer.java]: Yep.

**Pull request closed without merging**